### PR TITLE
Documentation: how to run FESTIM in parallel #856

### PR DESCRIPTION
**Pull request by XinShen-CHN**
## Proposed changes

Documentation: how to run FESTIM in parallel #856.

## Types of changes

What types of changes does your code introduce to FESTIM?

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code refactoring
- [x] Documentation Update (if none of the other choices apply)
- [ ] New tests

## Checklist

- [ ] Black formatted
- [ ] Unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: docs/source/userguide/beginners_guide.rst [hunk 1]
```diff
@@ -12,6 +12,12 @@ New users should familiarise themselves with `Conda <https://anaconda.org/>`_.
 
 FESTIM is under version control with `git <https://git-scm.com/>`_. Even though users don't need git to install FESTIM, it is convenient to understand how it works. You can find `git tutorials <https://git-scm.com/doc>`_ to help you getting started. The `FESTIM source code <https://github.com/festim-dev/FESTIM>`_ is hosted on GitHub. Signing up for free to GitHub is recommended to receive the latest updates, follow the development and submit `bug reports <https://github.com/festim-dev/FESTIM/issues/new/choose>`_.
 
+FESTIM can be run in parallel on N cores using the command: 
+
+.. testcode::
```
Comment (RemDelaporteMathurin) [inline docs/source/userguide/beginners_guide.rst:17]:
```suggestion
.. code::
```

If you use ``testcode`` this code block will be tested as if it was a python code block. This obviously won't work and this is why the documentation test is failing. 
Simply replace this by a normal code block.

**Review by RemDelaporteMathurin — CHANGES_REQUESTED** [review]
Thanks @XinShen-CHN for adding this!

I see that the documentation test is failing. See my comment below.

@KulaginVladimir do you think this should be at the bottom of this page in "Your first FESTIM script"?

**Comment by KulaginVladimir** [comment]
@RemDelaporteMathurin, I don't. Since parallel computing doesn't always result in faster computation, I wouldn't put it in the "your first FESTIM script". 

Personally, I'd make a separate page that could be filled with some additional info/hints/warnings later. 

We could also make a demo to show when parallel computing might speed up a simulation.

**Comment by XinShen-CHN** [comment]
> Thanks @XinShen-CHN for adding this!
> 
> I see that the documentation test is failing. See my comment below.

Thank you Remi!

**Comment by RemDelaporteMathurin** [comment]
> Personally, I'd make a separate page that could be filled with some additional info/hints/warnings later.

@XinShen-CHN could you make a new User Guide page "Running in parallel" and move the content there instead?

**Review by RemDelaporteMathurin — CHANGES_REQUESTED** [review]
Xin, you need to:
- provide the correct extension for the new file (.rst). Also, avoid blank spaces and uppercase letters in filenames. `running_in_parallel.rst` would be an appropriate filename.
- add this new file in `docs/source/userguide/index.rst` if you want it to be counted.

I advise you to try and build the documentation locally on your computer to be able to inspect your changes. Here you will see that these lines don't appear anywhere in the [automatically-built documentation](https://festim--871.org.readthedocs.build/en/871/userguide/index.html).

**Comment by XinShen-CHN** [comment]
> Xin, you need to:
> 
> * provide the correct extension for the new file (.rst). Also, avoid blank spaces and uppercase letters in filenames. `running_in_parallel.rst` would be an appropriate filename.
> * add this new file in `docs/source/userguide/index.rst` if you want it to be counted.
> 
> I advise you to try and build the documentation locally on your computer to be able to inspect your changes. Here you will see that these lines don't appear anywhere in the [automatically-built documentation](https://festim--871.org.readthedocs.build/en/871/userguide/index.html).

I see. I'll try to fix it.

Sorry for the delays. I was on a vacation.

**Comment by RemDelaporteMathurin** [comment]
Hi @XinShen-CHN do you need assistance on this?

**Comment by XinShen-CHN** [comment]
> Hi @XinShen-CHN do you need assistance on this?

Yes, I failed to build the documentation locally.

File: docs/source/userguide/running_in_parallel.rst [hunk 2]
```diff
@@ -0,0 +1,5 @@
+FESTIM can be run in parallel on N cores using the command:
```
Comment (RemDelaporteMathurin) [inline docs/source/userguide/running_in_parallel.rst:1]:
```suggestion
# Running in parallel

FESTIM can be run in parallel on N cores using the command: 
```

**Review by RemDelaporteMathurin — COMMENTED** [review]
Hi @XinShen-CHN thanks for these changes. I can assist you in building the docs locally. In the meantime, here's a final suggestion (I'll commit): we need to have a title (with # ) in this page to avoid this:

image

File: docs/source/userguide/running_in_parallel.rst [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# Running in parallel
```
Comment (RemDelaporteMathurin) [inline docs/source/userguide/running_in_parallel.rst:1]:
```suggestion
============
Running in parallel
============
```

File: docs/source/userguide/running_in_parallel.rst [hunk 4]
```diff
@@ -0,0 +1,9 @@
+============
+Running in parallel
+============
+
+FESTIM can be run in parallel on N cores using the command: 
+
+.. code::
+    
+    mpirun -np N python3 your_FESTIM_script.py.
```
Comment (RemDelaporteMathurin) [inline docs/source/userguide/running_in_parallel.rst:9]:
```suggestion
    mpirun -np N python3 your_FESTIM_script.py
```

**Review by RemDelaporteMathurin — APPROVED** [review]

**Comment by XinShen-CHN** [comment]
> Hi @XinShen-CHN thanks for these changes. I can assist you in building the docs locally. In the meantime, here's a final suggestion (I'll commit): we need to have a title (with # ) in this page to avoid this:
> 
> image

Remi, thank you so much!

**Pull request merged**